From 150a25ed86f3774d0d1ea9dbc307a8aa98822d39 Mon Sep 17 00:00:00 2001
From: Chao Guo <chao.guo@nxp.com>
Date: Thu, 15 Dec 2022 13:06:12 +0800
Subject: [PATCH] fixate caps to NV12 for 8qm, 8qm encoder only support NV12
 input

unlike 20.04, on 22.04 release need fix the caps of video_source_filter

Signed-off-by: Chao Guo <chao.guo@nxp.com>
---
 libcheese/cheese-camera.c | 9 ++++++---
 1 file changed, 6 insertions(+), 3 deletions(-)

diff --git a/libcheese/cheese-camera.c b/libcheese/cheese-camera.c
index 626fc1d6..40c8a3bb 100644
--- a/libcheese/cheese-camera.c
+++ b/libcheese/cheese-camera.c
@@ -746,15 +746,18 @@ cheese_camera_set_new_caps (CheeseCamera *camera)
 
   if (!gst_caps_is_empty (caps))
   {
+    /* GStreamer >= 1.1.4 expects fully-specified video-capture-source caps. */
+    caps = gst_caps_fixate (caps);
+    if (gst_caps_is_fixed (caps)) {
+      gst_caps_set_simple (caps, "format", G_TYPE_STRING, "NV12", NULL);
+    }
+
     GST_INFO_OBJECT (camera, "SETTING caps %" GST_PTR_FORMAT, caps);
     g_object_set (gst_bin_get_by_name (GST_BIN (priv->video_source),
                   "video_source_filter"), "caps", caps, NULL);
     g_object_set (priv->camerabin, "viewfinder-caps", caps,
                   "image-capture-caps", caps, NULL);
 
-    /* GStreamer >= 1.1.4 expects fully-specified video-capture-source caps. */
-    caps = gst_caps_fixate (caps);
-
     g_object_set (priv->camerabin, "video-capture-caps", caps, NULL);
 
     gst_caps_unref (caps);
-- 
2.17.0

